test-data/unit/check-classes.test

```diff
@@ -1723,3 +1723,61 @@ def foo(c: Type[C], d: Type[D]) -> None:
 [out]
 main: note: In function "foo":
 main:7: error: Revealed type is 'builtins.list[Type[__main__.B]]'
+
+[case testTypeMatchesOverloadedFunctions]
+from typing import Type, overload, cast
+class User: pass
+@overload
+def f(a: object) -> None: pass
+@overload
+def f(a: int) -> None: pass
+f(User)
+f(cast(Type[User], User))
```

Comment:
It would be nice to verify that mypy selects the correct overload variant in both cases here (and in other tests). A possibly pattern to do that:
```
x = f(...)
reveal_type(x) # E: ...
```
(Fill in the error message produced by reveal_type() on the `# E:` line.)